iot-e2e-tests/common/src/main/java/com/microsoft/azure/sdk/iot/common/serviceclient/JobClientCommon.java

```diff
@@ -67,15 +67,28 @@ private JobResult queryDeviceJobResult(String jobId, JobType jobType, JobStatus
         while(jobClient.hasNextJob(query))
         {
             jobResult = jobClient.getNextJob(query);
-            if(jobResult.getJobId().equals(jobId) &&
-                    (jobResult.getJobType() == jobType) &&
-                    (jobResult.getJobStatus() == jobStatus))
+            if(jobResult.getJobId().equals(jobId))
```

Comment:
use assertEquals()??

Reply:
The assertion failure takes place below when we manually throw a new AssertionError instead

Reply:
It's just that it will be cleaner to have 3 assertEquals() in place of this nested if else.   My 2 cents

---
In reply to: [227915481](https://github.com/Azure/azure-iot-sdk-java/pull/352#discussion_r227915481) [](ancestors = 227915481)